sos/report/plugins/microshift.py

```diff
@@ -0,0 +1,173 @@
+        res = self.exec_cmd('microshift get namespaces -o custom-columns=NAME:.metadata.name --no-headers')
+        if res['status'] == 0:
+            return self._reduce_namespace_list(res['output'].split('\n'))
+        return []
+
+    def _collect_cluster_resources(self):
+        """Collect cluster-level (non-namespaced) resources
+        """
+        global_resources = [
+            'apiservices',
+            'certificatesigningrequests',
+            'clusterrolebindings',
+            'clusterroles',
+            'componentstatuses',
+            'csidrivers',
+            'csinodes',
+            'customresourcedefinitions',
+            'flowschemas',
+            'ingressclasses',
+            'logicalvolumes',
+            'mutatingwebhookconfigurations',
+            'nodes',
+            'persistentvolumes',
+            'priorityclasses',
+            'prioritylevelconfigurations',
+            'rangeallocations',
+            'runtimeclasses',
+            'securitycontextconstraints',
+            'selfsubjectaccessreviews',
+            'selfsubjectrulesreviews',
+            'storageclasses',
+            'subjectaccessreviews',
+            'tokenreviews',
+            'validatingwebhookconfigurations',
+            'volumeattachments'
+        ]
+
+        _filtered_resources = []
+
+        # Filter out any resources that might not exist.
+        for resource in global_resources:
+            res = self.exec_cmd("microshift get %s" % resource)
+            if res['status'] == 0:
+                _filtered_resources.append(resource)
+
+        _subdir = self.get_cmd_output_path(make=False)
+        self.exec_cmd(
+            'microshift inspect --kubeconfig %s --dest-dir %s %s' % (
+                self.get_option('kubeconfig'),
+                _subdir,
+                ",".join(_filtered_resources)))
+
+    def _collect_named_resources(self):
+        """Collect namespaced resources (including pod logs) from designated
+        namespaces
+        """
+        _namespaces = ['ns/%s' % n for n in self._get_namespaces()]
+        _subdir = self.get_cmd_output_path(make=False)
+        res = self.exec_cmd(
+            'microshift inspect --kubeconfig %s --dest-dir %s %s' % (
+                self.get_option('kubeconfig'),
+                _subdir,
+                " ".join(_namespaces)))
+
+    def setup(self):
+        """The setup() phase of this plugin will first gather system
+        information and then iterate through all default namespaces, and/or
+        those specified via the `add-namespaces` and `only-namespaces` plugin
+        options. Both of these options accept shell-style regexes.
+
+        Output format for this function is based on `oc adm inspect` command,
+        which is used to retrieve all API resources from the cluster.
+        """
+
+        self.add_forbidden_path('/var/lib/microshift')
+        self.add_journal('microshift')
+        self.add_service_status('microshift')
```

Comment:
Since `microshift` is defined in the `services` attr, these are unnecessary as these collections are automatically added for any service defined for a plugin.

Reply:
Done